pandas/core/dtypes/cast.py

```diff
@@ -615,8 +616,18 @@ def astype_nansafe(arr, dtype, copy=True, skipna=False):
 
     # dispatch on extension dtype if needed
     if is_extension_array_dtype(dtype):
-        return dtype.construct_array_type()._from_sequence(
-            arr, dtype=dtype, copy=copy)
+        try:
+            array_type = dtype.construct_array_type()
+        except AttributeError:
+            dtype = pandas_dtype(dtype)
+            array_type = dtype.construct_array_type()
+        try:
+            # use _from_sequence_of_strings if the class defines it
+            return array_type._from_sequence_of_strings(arr,
```

Comment:
why is `_from_sequence_of_strings` being used here? As this is for general casting and not necessarily string parsing?

Reply:
It is used by the `python` engine, `PythonParser._cast_types`. When it gets to this point it is casting the strings (from the CSV) into `ndarrays`.

Reply:
why is this changed at all?

Reply:
@kprestel don't resolve something unless you respond. you haven't answered the question at all.